devdoc/doublylinkedlist_requirements.md

```diff
@@ -104,3 +134,48 @@ extern PDLIST_ENTRY DList_RemoveHeadList(PDLIST_ENTRY listHead);
 **SRS_DLIST_06_013: [** DList_RemoveHeadList shall return listHead if that's the only item in the list. **]**
 
 Note: The Flink & Blink of the returned PDLIST_ENTRY shall be undefined.
+
+### DList_Find
+```c
+MOCKABLE_FUNCTION(, DLIST_FIND_RESULT, DList_Find, PDLIST_ENTRY, listHead, DLIST_MATCH_FUNCTION, matchFunction, const void*, matchContext, PDLIST_ENTRY*, foundEntry);
+```
+
+**S_R_S_DLIST_43_001: [** `DList_Find` shall call `matchFunction` on each entry in the list defined by `listHead` along with `matchContext`. **]**
```

Comment:
>S_R_S_DLIST_43_001 [](http://example.com/codeflow?start=2&length=18)

all values for all parameters are valid? What happens if foundEntry is `NULL`?

Or if listHead is `NULL`?

Reply:
The spec says: "no input error checking is provided for this set of APIs". 

So I didn't add these checks for the new helpers either. Undefined behavior if input is invalid.